client/my-sites/sharing/buttons/preview-action.jsx

```diff
@@ -1,52 +1,44 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import { omit, startsWith, endsWith } from 'lodash';
 
-module.exports = React.createClass( {
-	displayName: 'SharingButtonsPreviewAction',
+const SharingButtonsPreviewAction = ( props ) => {
+	const { active, position, icon, children } = props;
+	const classes = classNames( 'sharing-buttons-preview-action', {
+		'is-active': active,
+		'is-top': startsWith( position, 'top' ),
+		'is-right': endsWith( position, '-right' ),
+		'is-bottom': startsWith( position, 'bottom' ),
+		'is-left': endsWith( position, '-left' )
+	} );
 
-	propTypes: {
-		active: React.PropTypes.bool,
-		position: React.PropTypes.oneOf( [
-			'top-left',
-			'top-right',
-			'bottom-left',
-			'bottom-right'
-		] ),
-		icon: React.PropTypes.string,
-		onClick: React.PropTypes.func
-	},
+	return (
+		<button type="button" className={ classes } { ...omit( props, [ 'active', 'position' ] ) }>
```

Comment:
I get your point - it can be confusing to receive props that a component is not expecting specifically.

However I think it is a bad idea to list all of these props. Simply because it can be a huge overkill - imagine if you have 5 different HTML tags in your component, and you list all of the possible attributes as props - that would be a huge list of props that nobody would probably use. So let's keep it as it is.